larsim/IonizationScintillation/ISCalcCorrelated.cxx

```diff
@@ -0,0 +1,146 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       ISCalcCorrelated
+// Plugin Type: algorithm
+// File:        ISCalcCorrelated.h and ISCalcCorrelated.cxx
+// Description: Interface to algorithm class for a specific calculation of
+//              ionization electrons and scintillation photons, based on 
+//              simple microphysics arguments to establish an anticorrelation
+//              between these two quantities.
+// Input: 'sim::SimEnergyDeposit'
+// Output: num of Photons and Electrons
+// May 2020 by W Foreman
+////////////////////////////////////////////////////////////////////////
+
+#include "larcore/CoreUtils/ServiceUtil.h"
+#include "larsim/IonizationScintillation/ISCalcCorrelated.h"
+
+namespace larg4
+{
+    //----------------------------------------------------------------------------
+    ISCalcCorrelated::ISCalcCorrelated()
+    {
+    }
+
+    //----------------------------------------------------------------------------
+    ISCalcCorrelated::~ISCalcCorrelated()
+    {
+    }
+
+    //----------------------------------------------------------------------------
+    void ISCalcCorrelated::Initialize()
+    {
+        std::cout << "IonizationAndScintillation/ISCalcCorrelated Initialize." << std::endl;
+        art::ServiceHandle<sim::LArG4Parameters const> LArG4PropHandle;
+
+        fSCE       = lar::providerFrom<spacecharge::SpaceChargeService>();
+        fLArProp   = lar::providerFrom<detinfo::LArPropertiesService>();
+        fDetProp   = lar::providerFrom<detinfo::DetectorPropertiesService>();
+
+        fScintYieldFactor  = 1.; // true scintillation yield will be got from LArProperties
+
+        //the recombination coefficient is in g/(MeVcm^2), but we report energy depositions in MeV/cm,
+        //need to divide Recombk from the LArG4Parameters service by the density of the argon we got above.
+        fRecombA          = LArG4PropHandle->RecombA();
+        fRecombk          = LArG4PropHandle->Recombk()/fDetProp->Density(fDetProp->Temperature());
+        fModBoxA          = LArG4PropHandle->ModBoxA();
+        fModBoxB          = LArG4PropHandle->ModBoxB()/fDetProp->Density(fDetProp->Temperature());
+        fUseModBoxRecomb  = (bool)LArG4PropHandle->UseModBoxRecomb();
+        fGeVToElectrons   = LArG4PropHandle->GeVToElectrons();
+        
+        // ionization work function
+        fWion             = 1./fGeVToElectrons * 1e3; // MeV
+
+        // ion+excitation work function (\todo: get from LArG4Parameters or LArProperties?)
+        fWph              = 19.5 * 1e-6; // MeV
+
+        return;
+    }
+
+    //----------------------------------------------------------------------------
+    void ISCalcCorrelated::Reset()
+    {
+        fEnergyDeposit           = 0.;
+        fNumScintPhotons         = 0.;
+        fNumIonElectrons         = 0.;
+        fScintillationYieldRatio = 0.;
```

Comment:
Any variables that are used in only one function should be defined in only that function--e.g. `fEnergyDeposit` need not be a class member variable as only `CalcIonAndScint` uses it.